### DISCUSSION TITLE
NextJS with SSR for Self-Service API

### DISCUSSION THREAD

**Discussion by quizmoon** [ory/kratos#2477] [Q&A] [answered]
Good day,

we are working on the NextJS project, in that project all of the pages require authentication and authorization control, so we are ended up to use SSR for all of them using getServerSideProps.

Now we are on the way to transform UI for Kratos Self-Services to use getServerSideProps, and found that not every of [Kratos Self-Service HTTP API](https://www.ory.sh/docs/reference/api) have "cookie" in Header Parameters, so we can not delivery Session Cookie to the call. 

For example, to process with [submitSelfServiceLogoutFlow](https://www.ory.sh/docs/reference/api#operation/submitSelfServiceLogoutFlow) there is no "cookie" in Header Parameters, as result we get 401. 

Of course we can set cookie in Axios configs, just trying to understand  if it was made on purpose or just skipped.

P.S. We would not like to open direct access to Kratos Self-Service URLS and would not like to use @ory/client in useEffects. 

Will be thankful for any hint, possible we choose wrong approach and there is better and secure ways.

Thank you.

**Comment by aeneasr** [comment, marked as answer]
Thank you for the question @quizmoon ! That is indeed a problem, it would be awesome if you could point out the methods that lack the cookie parameter and we'll add the cookie parameter to those

**Reply by vinckr** [reply]
Hey @quizmoon,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
